### PR TITLE
Users

### PR DESCRIPTION
**Pull request by erinlimbo**
Users are now identifiable in their posts

### CODE REVIEW

**Comment by erinlimbo** [comment]
I decided to also incorporate Date time so this pull request isnt ready yet

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -46,10 +48,16 @@
   /** The max amount of comments that will be displayed. */
   private int maxComments = 40;
 
+  /** User service that contains the information of the current user. */
+  private UserService userService = UserServiceFactory.getUserService();
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: private final

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -74,10 +82,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO : Incorporate users for author, and Date for timeStamp
+
+    User currentUser = userService.getCurrentUser();
+
     String comment = request.getParameter("comment-text");
-    String author = "anonymous";
-    String timeStamp = "11/23/2000";
+    String author = currentUser.getEmail().split("@", 2)[0];
+    Date timeStamp = new Date();
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:91]: nit: add comment here to say that this gives the current time

File: portfolio/src/main/java/com/google/sps/servlets/MemeHandler.java [hunk 3]
```diff
@@ -41,10 +46,16 @@
   /** The max amount of memes that will be displayed. */
   private int maxMemes = 40;
 
+  /** User service that contains the information of the current user. */
+  private UserService userService = UserServiceFactory.getUserService();
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/MemeHandler.java]: private final

File: portfolio/src/main/java/com/google/sps/servlets/MemeHandler.java [hunk 4]
```diff
@@ -71,17 +82,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     /** Receive a meme and upload its data to the datastore. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    User currentUser = userService.getCurrentUser();
 
+    String author = currentUser.getEmail().split("@", 2)[0];
     String description = request.getParameter("message");
-
     String imageUrl = getUploadedFileUrl(request, "file");
+    Date timeStamp = new Date();
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/MemeHandler.java:91]: nit: same thing here

**Review by laskdaf — APPROVED** [review]
a few small comments. please address before merging. nice work :)

**Pull request merged**